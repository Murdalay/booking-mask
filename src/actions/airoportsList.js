// @flow
/**
 * @module Actions/airoportsList
 * @desc airoportsList Actions
 */
import { createActions } from 'redux-actions';

import { ActionTypes } from 'constants/index';

export const { airoportsListGet } = createActions({
  [ActionTypes.AIROPORTS_LIST_GET]: () => ({}),
});
