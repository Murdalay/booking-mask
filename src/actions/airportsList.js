// @flow
/**
 * @module Actions/airportsList
 * @desc airportsList Actions
 */
import { createActions } from 'redux-actions';

import { ActionTypes } from 'constants/index';

export const { airportsListGet } = createActions({
  [ActionTypes.AIRPORTS_LIST_GET]: () => ({}),
});
