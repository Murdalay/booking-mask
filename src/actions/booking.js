// @flow
/**
 * @module Actions/Booking
 * @desc Booking Actions
 */
import { createActions } from 'redux-actions';

import { ActionTypes } from 'constants/index';

export const {
  bookingChangePassangersNumber: changePassangersNumber,
  bookingChangeFrom: changeFrom,
  bookingChangeTo: changeTo,
  bookingChangeFlightDates: changeFlightDates,
  bookingSubmit: submit,
  bookingUnsetInvalid: unsetInvalid,
  bookingTogglePassangersDropdown: togglePassangersDropdown,
} = createActions({
  [ActionTypes.BOOKING_GET_REPOS]: (query: string) => ({ query }),
  [ActionTypes.BOOKING_CHANGE_PASSANGERS_NUMBER]: (data: {
    adults: number,
    children: number,
    infants: number,
  }) => ({ data }),
  [ActionTypes.BOOKING_CHANGE_FROM]: data => ({ code: data }),
  [ActionTypes.BOOKING_CHANGE_TO]: data => ({ code: data }),
  [ActionTypes.BOOKING_CHANGE_FLIGHT_DATES]: (data: { startDate: string, endDate: string }) => data,
  [ActionTypes.BOOKING_SUBMIT]: () => ({}),
  [ActionTypes.BOOKING_UNSET_INVALID]: () => ({}),
  [ActionTypes.BOOKING_TOGGLE_PASSANGERS_DROPDOWN]: () => ({}),
});
