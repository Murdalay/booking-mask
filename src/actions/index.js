export * from './app';
export * from './booking';
export * from './airoportsList';
