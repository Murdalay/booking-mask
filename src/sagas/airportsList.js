/**
 * @module Sagas/airportsList
 * @desc airportsList
 */

import { all, call, put, takeLatest } from 'redux-saga/effects';
import { mimicRequest } from 'modules/client';
import { ActionTypes } from 'constants/index';

/**
 * Get Repos
 */
export function* getAiroportsList() {
  try {
    const response = yield call(mimicRequest, 'https//sample-url');
    yield put({
      type: ActionTypes.AIRPORTS_LIST_GET_SUCCESS,
      payload: { data: response },
    });
  } catch (err) {
    /* istanbul ignore next */
    yield put({
      type: ActionTypes.AIRPORTS_LIST_GET_FAILURE,
      payload: err,
    });
  }
}

/**
 * airportsList Sagas
 */
export default function* root() {
  yield all([takeLatest(ActionTypes.AIRPORTS_LIST_GET, getAiroportsList)]);
}
