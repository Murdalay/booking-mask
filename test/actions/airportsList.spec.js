import { airportsListGet } from 'actions/airportsList';

describe('App', () => {
  it('airportsListGet should return an action', () => {
    expect(airportsListGet()).toMatchSnapshot();
  });
});
