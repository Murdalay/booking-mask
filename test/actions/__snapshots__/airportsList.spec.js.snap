// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App airportsListGet should return an action 1`] = `
Object {
  "payload": Object {},
  "type": "AIRPORTS_LIST_GET",
}
`;
